src/modules/endpoint/Endpoint.js

```diff
@@ -98,12 +94,12 @@ class Endpoint {
     return `${this.stateName}Payload`
   }
 
-  requestAction (urlParams = {}) {
+  requestAction (router) {
```

Comment:
Why did you change this to router? All implementations of endpoints just use the params object in that 'router'.
I don't really want a dependency from the router to the endpoint API because the router is likely to change in the future.

Reply:
We could introduce an ParamContext or LocationContext which holds the params and offers methods for accessing them.

Reply:
Or we change 'router' to 'state' and just pass the whole state. This would also allow to include the state in the url. Which is basically what this feature would be for. So we would have a mapStateToUrl function.

Reply:
you're right, so only params or the whole state....which would you prefer? the state could be big...but on the other hand for the extras I need the state